libreoffice build failure on master

**Describe the bug**

Building `libreoffice-still` and therefore also 'libreoffice` fails on `master`.

**To Reproduce**
Steps to reproduce the behavior:
1. `nix build -f https://github.com/NixOS/nixpkgs/archive/cc7eaae.tar.gz libreoffice`
2. Build fails with:
```log
  checking for gtk+-2.0 >= 2.18.0 gdk-pixbuf-2.0 >= 2.2 ... no
  configure: error: requirements to build the gtk-plugin not met. Use --disable-gtk or install the missing packages
```
[libreoffice-6.3.5.2.log](https://github.com/NixOS/nixpkgs/files/4734749/libreoffice-6.3.5.2.log)

**Expected behavior**
The build succeeds.

**Additional context**
Bisecting show da540dbb7019e6a7f87ab28dbd6da5d16f13c32d (#89395) as the first failing commit.
```
da540dbb7019e6a7f87ab28dbd6da5d16f13c32d is the first bad commit
commit da540dbb7019e6a7f87ab28dbd6da5d16f13c32d
Author: Peter Hoeg <peter@hoeg.com>
Date:   Wed Jun 3 15:15:17 2020 +0800

    libreoffce-qt: init at 6.4.3.2

 pkgs/applications/office/libreoffice/default.nix | 76 ++++++++++++++++++------
 pkgs/top-level/all-packages.nix                  |  8 +++
 2 files changed, 66 insertions(+), 18 deletions(-)
```

**Notify maintainers**
@7c6f434c @peterhoeg

**Metadata**
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.6.15, NixOS, 20.09pre227577.135073a87b7 (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.5`
 - channels(root): `"nixos-20.09pre228204.467ce5a9f45"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
- libreoffice-still
- libreoffice
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/applications/office/libreoffice/default.nix
+++ b/pkgs/applications/office/libreoffice/default.nix
@@ -96,9 +96,9 @@ in (mkDrv rec {
   # are in multiple directories due to each having their own derivation.
   postPatch = let
     inc = e: path:
-      "${e.dev}/include/KF5/${path}";
+      "${lib.getDev e}/include/KF5/${path}";
     libs = list:
-      lib.concatMapStringsSep " " (e: "-L${e.out}/lib") list;
+      lib.concatMapStringsSep " " (e: "-L${lib.getLib e}/lib") list;
   in ''
     substituteInPlace shell/source/unix/exec/shellexec.cxx \
       --replace /usr/bin/xdg-open ${if kdeIntegration then "kde-open5" else "xdg-open"}
@@ -377,7 +377,7 @@ in (mkDrv rec {
     "--enable-kf5"
     "--enable-qt5"
     "--enable-gtk3-kde5"
-  ];
+  ] ++ lib.optional (variant == "still") "--disable-gtk"; # disables GTK2, GTK3 is still there
 
   checkPhase = ''
     make unitcheck

--- a/pkgs/applications/office/libreoffice/default.nix
+++ b/pkgs/applications/office/libreoffice/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, fetchurl, fetchpatch, pam, python3, libxslt, perl, ArchiveZip, gettext
+{ stdenv, fetchurl, fetchpatch, lib, pam, python3, libxslt, perl, ArchiveZip, gettext
 , IOCompress, zlib, libjpeg, expat, freetype, libwpd
 , libxml2, db, curl, fontconfig, libsndfile, neon
 , bison, flex, zip, unzip, gtk3, libmspack, getopt, file, cairo, which
@@ -29,7 +29,6 @@ let
 
   inherit (primary-src) major minor subdir version;
 
-  lib = stdenv.lib;
   langsSpaces = lib.concatStringsSep " " langs;
 
   mkDrv = if kdeIntegration then mkDerivation else stdenv.mkDerivation;
@@ -377,7 +376,7 @@ in (mkDrv rec {
     "--enable-kf5"
     "--enable-qt5"
     "--enable-gtk3-kde5"
-  ] ++ lib.optional (variant == "still") "--disable-gtk"; # disables GTK2, GTK3 is still there
+  ] ++ lib.optional (lib.versionOlder version "6.4") "--disable-gtk"; # disables GTK2, GTK3 is still there
 
   checkPhase = ''
     make unitcheck

